Java/src/org/imos/ddb/JDBCDDB.java

```diff
@@ -33,160 +33,193 @@
-        for (Field f : fields) {
-          
-          if (f.isSynthetic()) continue;
-          
-          Object o = rs.getObject(f.getName());
-
-          //all numeric values must be doubles
-          if (o instanceof Integer) {
-            o = (double)((Integer)o).intValue();
-
-            //Hack to accommodate DeploymentId and FieldTripID 
-            //types of number or text. Don't tell anyone
-            if (f.getType() == String.class)
-              o = o.toString();
-          }
-            
-          f.set(instance, o);
-        }
-      }
-    }
-    
-    //always close db connection
-    finally {try {conn.close();} catch (Exception e) {}}
-    
-    return results;
-  }
+	/**
+	 * Create a JDBC DDB object using the specified JDBC driver and database
+	 * connection.
+	 * 
+	 * @param driver
+	 *            Class name of JDBC database driver
+	 * @param connection
+	 *            Database connection string, must include user and password if
+	 *            required by the database.
+	 * @throws ClassNotFoundException
+	 *             If the specified Database driver can't be found
+	 * @throws SQLException
+	 *             If an attempt to open a connection to the database fails
+	 */
+	protected JDBCDDB(String driver, String connection, String user, String password) throws ClassNotFoundException, SQLException
+	{
+		this.driver = driver;
+		this.connection = connection;
+		this.user = user;
+		this.password = password;
+
+		// Test connection - throws exception at creation if can't connect
+		Class.forName(driver);
+
+		Connection conn = DriverManager.getConnection(connection, user, password);
+		
+		conn.close();
+	}
+
+	/**
+	 * Executes the given query, in the form:
+	 * 
+	 * select * from [tableName] where [fieldName] = '[fieldValue]'
+	 * 
+	 * The rows are converted into object equivalents of the table, and returned
+	 * in a List. If fieldName is null, the where clause is omitted, thus the
+	 * entire table is returned.
+	 * 
+	 * @param tableName
+	 *            The table to read.
+	 * 
+	 * @param fieldName
+	 *            the name of the query field.
+	 * 
+	 * @param fieldValue
+	 *            the query field value.
+	 * 
+	 * @throws Exception
+	 *             on any error.
+	 */
+	public List<Object> executeQuery(String tableName, String fieldName, Object fieldValue) throws Exception
+	{
+
+		Connection conn = null;
+		List<Object> results = null;
+
+		try
+		{
+
+			// create ODBC database connection
+			Class.forName(driver);
+
+			conn = DriverManager.getConnection(connection, user, password);
+
+			results = new ArrayList<Object>();
+
+			// build the query
+			String query = "SELECT * FROM " + tableName;
+			if (fieldName != null)
+			{
+
+				if (fieldValue == null)
+					throw new Exception("a fieldValue must be provided");
+
+				// wrap strings in quotes
+				if (fieldValue instanceof String)
+					query += " WHERE \"" + fieldName + "\" = '" + fieldValue + "'";

```

Comment:
Hmm, what does not work, does it not find the fieldnames, maybe an update to the .txt files so the names match the schema in the database.
Use the Test class to see what the fieldnames case is in the database.